connectors/agentic-ai/src/main/java/io/camunda/connector/agenticai/aiagent/agent/AiAgentRequestHandlerImpl.java

```diff
@@ -0,0 +1,242 @@
+import io.camunda.connector.agenticai.model.message.content.DocumentContent;
+import io.camunda.connector.agenticai.model.message.content.TextContent;
+import io.camunda.connector.agenticai.model.tool.ToolCallProcessVariable;
+import io.camunda.connector.agenticai.model.tool.ToolDefinition;
+import io.camunda.connector.api.error.ConnectorException;
+import io.camunda.connector.api.outbound.OutboundConnectorContext;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import org.apache.commons.lang3.StringUtils;
+
+public class AiAgentRequestHandlerImpl implements AiAgentRequestHandler {
+
+  private static final int DEFAULT_MAX_MODEL_CALLS = 10;
+  private static final int DEFAULT_MAX_MEMORY_MESSAGES = 20;
+
+  private static final String ERROR_CODE_NO_USER_MESSAGE_CONTENT = "NO_USER_MESSAGE_CONTENT";
+  private static final String ERROR_CODE_WAITING_FOR_TOOL_INPUT_EMPTY_RESULTS =
+      "WAITING_FOR_TOOL_INPUT_EMPTY_RESULTS";
+  private static final String ERROR_CODE_TOOL_CALL_RESULTS_ON_EMPTY_CONTEXT =
+      "TOOL_CALL_RESULTS_ON_EMPTY_CONTEXT";
+  private static final String ERROR_CODE_MAXIMUM_NUMBER_OF_MODEL_CALLS_REACHED =
+      "MAXIMUM_NUMBER_OF_MODEL_CALLS_REACHED";
+
+  private final AdHocToolsSchemaResolver schemaResolver;
+  private final AiFrameworkAdapter<?> framework;
+
+  public AiAgentRequestHandlerImpl(
+      AdHocToolsSchemaResolver schemaResolver, AiFrameworkAdapter<?> framework) {
+    this.schemaResolver = schemaResolver;
+    this.framework = framework;
+  }
+
+  @Override
+  public AgentResponse handleRequest(OutboundConnectorContext context, AgentRequest request) {
+    final AgentRequest.AgentRequestData requestData = request.data();
+    AgentContext agentContext = initializeAgentContext(context, request);
+
+    // set up memory and load from context if available
+    final var runtimeMemory =
+        new MessageWindowRuntimeMemory(
+            Optional.ofNullable(requestData.memory())
+                .map(MemoryConfiguration::maxMessages)
+                .orElse(DEFAULT_MAX_MEMORY_MESSAGES));
+
+    final var conversationStore = new InProcessConversationStore();
+    conversationStore.loadIntoRuntimeMemory(context, agentContext, runtimeMemory);
+
+    // check configured limits
+    checkLimits(requestData, agentContext);
+
+    // update memory with system message + new user messages/tool call responses
+    addSystemPromptIfNecessary(runtimeMemory, requestData);
+    addUserMessagesFromRequest(agentContext, runtimeMemory, requestData);
+
+    // call framework with memory
+    AiFrameworkChatResponse<?> frameworkChatResponse =
+        framework.executeChatRequest(request, agentContext, runtimeMemory);
+    agentContext = frameworkChatResponse.agentContext();
+
+    final var assistantMessage = frameworkChatResponse.assistantMessage();
+    runtimeMemory.addMessage(assistantMessage);
```

Comment:
This is where we interact with the framework (Langchain4J): we send the messages in our format into the framework and expect a response with the assistant message and tool calls already being converted to our format.